src/text_formatter/allowed_Expression_Detectors.ts

```diff
@@ -0,0 +1,60 @@
+// --------------------------------------
+// @author: Ronja K.
+// ---------------------------------------
+
+
+// ---------CurrentExpressionDetector------------
+// ---------every element have their own detect-function--------
+export function detectExpressionElements(input: string) {
+	input = detectBracketLeft(input);
+	input = detectBracketRight(input);
+	input = detectEquivalence(input);
+	input = detectImplication(input);
+	input = detectImplication(input);
+	input = detectNegation(input);
+	input = detectConjunction(input);
+	input = detectDisjunction(input);
+	return input;
```

Comment:
You are returning a string here, with symbols replaced by tokens, separated by spaces, right? This is not an especially handy return-type. Consider returning an `Array<Token>` instead, with `type Token = "bracketLeft" | "bracketRight" | "equivalence" |...` and specify that return type explicitly on the function!